OpenEdXMobile/src/main/java/org/edx/mobile/util/TimeZoneUtils.java

```diff
@@ -3,606 +3,30 @@
 import android.support.annotation.NonNull;
 import android.text.TextUtils;
 
-import java.util.HashMap;
+import java.util.Date;
 import java.util.TimeZone;
 
 public class TimeZoneUtils {
     /**
-     * Returns the region's abbreviation based on the time zone id provided.
+     * Returns the region's abbreviation based on the time zone provided.
      *
-     * @param timeZoneId Id of time zone.
+     * @param timeZone
      * @return Region's abbreviation of the time zone.
      */
-    public static String getTimeZoneAbbreviation(@NonNull String timeZoneId) {
-        String timeZoneAbbr;
+    public static String getTimeZoneAbbreviation(@NonNull TimeZone timeZone) {
         if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
-            try {
-                timeZoneAbbr = android.icu.util.TimeZone.getRegion(timeZoneId);
-                if (!TextUtils.isEmpty(timeZoneAbbr)) {
-                    return timeZoneAbbr;
-                }
-            } catch (IllegalArgumentException e) {
-                // id is not a known system ID.
+            android.icu.util.TimeZone tz = android.icu.util.TimeZone.getTimeZone(timeZone.getID());
+            final String tzAbbr = tz.getDisplayName(timeZone.inDaylightTime(new Date()),
```

Comment:
Why aren't we using `java.util.TimeZone#useDaylightTime` function instead?

Reply:
`inDaylightTime` queries if the given date is in Daylight Saving Time of particular time zone which is actually required while `useDaylightTime` just tells either the time zone calculation of particular time zone is based on day light saving or not.

Reply:
Got it!